src/main/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigClientImpl.java

```diff
@@ -94,8 +95,9 @@ public RemoteConfigTemplate getTemplate() throws FirebaseRemoteConfigException {
     HttpRequestInfo request = HttpRequestInfo.buildGetRequest(remoteConfigUrl)
             .addAllHeaders(COMMON_HEADERS);
     IncomingHttpResponse response = httpClient.send(request);
-    RemoteConfigTemplate parsed = httpClient.parse(response, RemoteConfigTemplate.class);
-    return parsed.setETag(getETag(response));
+    TemplateResponse templateResponse = httpClient.parse(response, TemplateResponse.class);
+    RemoteConfigTemplate template = templateResponse.toPublicType();
```

Comment:
`toRemoteConfigTemplate()`

In our chats I used `toPublicType()` and `toResponseType()` as placeholders. Use the typical Java conventions to name these methodds.